cider/exec/nextgen/jitlib/base/JITValueOperations.h

```diff
@@ -0,0 +1,62 @@
+#ifndef JITLIB_BASE_JITVALUEOPERATIONS_H
+#define JITLIB_BASE_JITVALUEOPERATIONS_H
+
+#include <type_traits>
+#include "exec/nextgen/jitlib/base/JITValue.h"
+#include "exec/nextgen/jitlib/llvmjit/LLVMJITFunction.h"
+
+namespace jitlib {
+template <typename T>
+struct is_jitvalue_convertable {
+  using NT = typename std::remove_reference<T>::type;
+  static constexpr bool v = std::is_arithmetic_v<T> || std::is_same_v<T, bool>;
+};
+template <typename T>
+inline constexpr bool is_jitvalue_convertable_v = is_jitvalue_convertable<T>::v;
+
+namespace op_utils {
+template <typename T>
+inline std::any castConstant(JITTypeTag target_type, T value) {
+  std::any ret;
+  switch (target_type) {
+    case BOOL:
+      return ret = static_cast<JITTypeTraits<BOOL>::NativeType>(value);
+    case INT8:
+      return ret = static_cast<JITTypeTraits<INT8>::NativeType>(value);
+    case INT16:
+      return ret = static_cast<JITTypeTraits<INT16>::NativeType>(value);
+    case INT32:
+      return ret = static_cast<JITTypeTraits<INT32>::NativeType>(value);
+    case INT64:
+      return ret = static_cast<JITTypeTraits<INT64>::NativeType>(value);
+    case FLOAT:
+      return ret = static_cast<JITTypeTraits<FLOAT>::NativeType>(value);
+    case DOUBLE:
+      return ret = static_cast<JITTypeTraits<DOUBLE>::NativeType>(value);
+    default:
+      return ret;
+  }
+}
+};  // namespace op_utils
+
+JITValuePointer operator+(JITValue& lh, JITValue& rh) {
+  return lh.add(rh);
```

Comment:
we should add `JITOperation : public JITValue` and then define all operation interfaces in `JITOperation`

Reply:
Emmm, seems some operators like ```+``` must be defined as a non-member function to support cases like ```1 + JITValue```.

Reply:
yep. Or we can only support `JITConstant(1) + JITValue`, not best but more clear.